### PR TITLE
Add Minio:

### PR DESCRIPTION
**Pull request by itayadler**
Minio is a distributed storage engine. Fully compatible
with S3 API.
Added `ExAws` the AWS Elixir API to use it locally.
This commit is the first step in adding support for
loading and saving packages.

### CODE REVIEW

File: server/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+version: '2'
+
+# starts 4 docker containers running minio server instances. Each
+# minio server's web interface will be accessible on the host at port
+# 9001 through 9004.
```
Comment (BlueHotDog) [inline server/docker-compose.yaml]: can you please explain why we want 4? or more than 1?

File: server/mix.exs [hunk 2]
```diff
@@ -45,6 +45,10 @@ defmodule BuffServer.MixProject do
       {:argon2_elixir, "~> 2.0"},
       {:comeonin, "~> 5.1"},
       {:joken, "~> 2.0"},
+      {:ex_aws, "~> 2.0"},
+      {:ex_aws_s3, "~> 2.0"},
+      {:poison, "~> 3.0"},
```
Comment (BlueHotDog) [inline server/mix.exs]: no chance they work with Jason? just having two json encoders is.. buh..

File: server/mix.exs [hunk 3]
```diff
@@ -45,6 +45,10 @@ defmodule BuffServer.MixProject do
...
+      {:hackney, "~> 1.9"},
```
Comment (BlueHotDog) [inline server/mix.exs]: who uses this?

File: server/docker-compose.yaml (hunk 1 above)
Comment (itayadler) [reply server/docker-compose.yaml]: Unfortunately Minio requires a minimum of 4, its a known thing in Minio

File: server/mix.exs (hunk 3 above)
Comment (itayadler) [reply server/mix.exs]: all those 4 are required by `ex_aws_s3`

File: server/mix.exs (hunk 2 above)
Comment (itayadler) [reply server/mix.exs]: let me try

**Pull request closed without merging**